### PR TITLE
feat(dart_frog_cli): add dart_frog_new_brick (middleware implementation)

### PR DESCRIPTION
**Pull request by renancaraujo**
## Status

**READY**

## Description

Continuation of  #623
ref: #260 

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: bricks/dart_frog_new/__brick__/{{filename}} [hunk 1]
```diff
@@ -1,3 +1,7 @@
 import 'package:dart_frog/dart_frog.dart';
 
-{{> route.dart}}
\ No newline at end of file
+{{#is_route}}
+{{> route.dart}}
+{{/is_route}}{{^is_route}}
+{{> middleware.dart}}
+{{/is_route}}
```
Comment (felangel) [inline bricks/dart_frog_new/__brick__/{{filename}}:7]:
```suggestion
{{/is_route}}{{is_middleware}}
{{> middleware.dart}}
{{/is_middleware}}
```

File: bricks/dart_frog_new/__brick__/{{~ middleware.dart }} [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Handler middleware(Handler handler) {
+  return handler.use(requestLogger());
```
Comment (felangel) [inline bricks/dart_frog_new/__brick__/{{~ middleware.dart }}:2]:
```suggestion
  // TODO: implement middleware
  return handler;
```

File: bricks/dart_frog_new/__brick__/{{filename}} (hunk 1 above)
Comment (renancaraujo) [reply bricks/dart_frog_new/__brick__/{{filename}}]: Added this and other stuff

File: bricks/dart_frog_new/__brick__/{{filename}} [hunk 3]
```diff
@@ -1,3 +1,6 @@
 import 'package:dart_frog/dart_frog.dart';
```
Comment (alestiago) [inline bricks/dart_frog_new/__brick__/{{filename}}:1]: Shall we include the space between the import and the first line to have a correctly formatted file?

```suggestion
import 'package:dart_frog/dart_frog.dart';

```

**Review by alestiago — APPROVED** [review]

File: bricks/dart_frog_new/__brick__/{{filename}} (hunk 3 above)
Comment (renancaraujo) [reply bricks/dart_frog_new/__brick__/{{filename}}:1]: This creates an unnecessary space, reverting it in the next pr

**Pull request merged**